src/from1DNMRVariables.ts

```diff
@@ -185,3 +221,21 @@ export function from1DNMRVariables(
   header += '##END=';
   return header;
 }
+
+function checkDescending(variables: NMR1DVariables) {
+  const xVariable = variables.x as MeasurementVariable<DoubleArray>;
+  const reverse =
+    xVariable.units !== 'time' && xVariable.data[0] < xVariable.data[1];
+
+  if (!reverse) return variables;
+
+  const newVariables = JSON.parse(
```

Comment:
This is extremely slow and should not be used. We need to make a zoom to discuss about this PR.